Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -142,18 +142,18 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
 }
 
 #if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
-void NetworkTaskCocoa::setCookieTransformForFirstPartyRequest(const WebCore::ResourceRequest& request)
+void NetworkTaskCocoa::setCookieTransformForThirdPartyRequest(const WebCore::ResourceRequest& request, IsRedirect isRedirect)
 {
     if (!request.isThirdParty())
         return;
 
-    ASSERT(!task()._cookieTransformCallback);
-    if (!!task()._cookieTransformCallback)
-        return;
-
     if (!Quirks::needsPartitionedCookies(request))
         return;
 
+    ASSERT_UNUSED(isRedirect, !task()._cookieTransformCallback || isRedirect == IsRedirect::Yes);
+    if (!!task()._cookieTransformCallback)
+        task()._cookieTransformCallback = nil;
```

Comment:
The !! is probably not necessary.  It also seems like this could just be an unconditional setting of the callback to nil, which would be equal or fewer ObjC calls.